### PR TITLE
debian: add eosknowledgecontent packages

### PR DESCRIPTION
**Pull request by mattdangerw**
https://phabricator.endlessm.com/T13418

### CODE REVIEW

File: debian/libeos-knowledge-content-0-0.install [hunk 1]
```diff
@@ -0,0 +1,2 @@
+usr/lib/libeos-knowledge-content-0.so.*
+usr/lib/gio/modules/libeknvfs.so
```
Comment (mattdangerw) [inline debian/libeos-knowledge-content-0-0.install]: @xjuan I didn't see the vfs library getting pulled into any existing packages, so I'm adding it here. Which I think is the right place for it anyway.

One thing I wasn't sure about was the gio-querymodules call here https://github.com/endlessm/eos-knowledge-lib/blob/286d25d82d7f1e35b2c7d42be0c359cd047ba1a3/Makefile.am#L302. If that's necessary it probably also needs to go in an post-install debian hook. Could you investigate and tag on a commit here in necessary?

Comment (xjuan) [inline debian/libeos-knowledge-content-0-0.install]: @mattdangerw yes we do need to call gio-querymodules in a post install hook
Its needed to update the cache of extension points, that way gio does not have to open the module each time to know which extension point it implements

**Comment by mattdangerw** [comment]
@xjuan Postinst added. I'm not sure if this will actually work as intended though. The gio module (on master and this branch) will get installed to `/usr/lib/gio/modules` but every existing module is in `/usr/lib/x86_64-linux-gnu/gio/modules` depending on the arch.

Does anyone know if ignoring the arch specific libdir is okay in this case? Or what would be needed to actually support multi-arch in knowledge lib like all those other packages installing gio modules do?

File: debian/control [hunk 2]
```diff
@@ -105,3 +106,51 @@ Depends: eos-knowledge-0 (= ${binary:Version}),
...
+ Package config and headers for the Endless Knowledge Content library
+
+Package: libeos-knowledge-content-0-doc
+Section: non-free/doc
+Architecture: any
+Depends: ${misc:Depends}
+Description: Endless Knowledge Content library documentation
+ HTML documentation for the Endless Knowledge Content library
+
+Package: gir1.2-eos-knowledge-content-0
```
Comment (ptomato) [inline debian/control]: This should probably be `gir1.2-eosknowledgecontent-0` to fit with other upstream names, like `gir1.2-gdkpixbuf-2.0` (I know we named `gir1.2-eos-shard-0` incorrectly)

File: debian/rules [hunk 3]
```diff
@@ -18,7 +18,7 @@ override_dh_install:
 
 # Build our documentation as well
 override_dh_auto_configure:
-	dh_auto_configure -- --enable-js-doc
+	dh_auto_configure -- --enable-js-doc --enable-gtk-doc
```
Comment (ptomato) [inline debian/rules]: It's a bit sad that we have to bring gtk-doc in here only to take it out later, but this does indeed seem like the best thing to do.

File: debian/control [hunk 4]
```diff
@@ -105,3 +106,51 @@ Depends: eos-knowledge-0 (= ${binary:Version}),
...
+Architecture: any
+Depends: ${misc:Depends},
+         libeos-knowledge-content-0-0 (= ${binary:Version})
+Description: Endless Knowledge Content library debugging symbols
+ Debugging symbols for the Endless Knowledge Content library
+
+Package: libeos-knowledge-content-0-dev
+Section: non-free/libdevel
+Architecture: any
+Depends: ${shlibs:Depends},
```
Comment (ptomato) [inline debian/control]: I don't think the `-dev` package needs `${shlibs:Depends}`, or does it?

File: debian/control [hunk 5]
```diff
@@ -105,3 +106,51 @@ Depends: eos-knowledge-0 (= ${binary:Version}),
...
+         ${misc:Depends},
+         libglib2.0-dev,
+         libeos-knowledge-content-0-0 (= ${binary:Version})
+Description: Endless Knowledge Content development files
+ Package config and headers for the Endless Knowledge Content library
+
+Package: libeos-knowledge-content-0-doc
+Section: non-free/doc
+Architecture: any
+Depends: ${misc:Depends}
```
Comment (ptomato) [inline debian/control]: You probably want `Suggests: devhelp` or some such thing here

**Comment by ptomato** [comment]
> every existing module is in `/usr/lib/x86_64-linux-gnu/gio/modules` depending on the arch.

Sounds like we should be doing that as well, how does it work in those modules' Debian packaging?

**Comment by ptomato** [comment]
I just found out we have a distro patch (taken from Debian) that enables the GIO modules path without the architecture as well. Probably for maximum compatibility it would be good to install the module in the right place though.

**Comment by mattdangerw** [comment]
Added multi arch support. Confirmed that everything is building, though I haven't actually tested the debians yet.

**Comment by xjuan** [comment]
CR pass

**Pull request merged**